libyul/optimiser/Suite.cpp

```diff
@@ -87,6 +87,41 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
 
+namespace
+{
+
+#ifdef OUTPUT_PERFORMANCE_METRICS
+void outputPerformanceMetrics(map<string, int64_t> const& _metrics)
+{
+	vector<pair<string, int64_t>> durations(_metrics.begin(), _metrics.end());
+	sort(
+		durations.begin(),
+		durations.end(),
+		[](pair<string, int64_t> const& _lhs, pair<string, int64_t> const& _rhs) -> bool
+		{
+			return _lhs.second < _rhs.second;
+		}
+	);
+	int64_t total = 0;
+	for (auto&& [step, count]: durations)
+		total += count;
```

Comment:
This bit was confusing. At first I thought you were counting the executions of each step and it wasn't clear to me that both `_metrics` and `durations` contain durations. I'd rename things here to make it clearer:
```suggestion
	int64_t totalDuration = 0;
	for (auto&& [step, duration]: sortedMetrics)
		total += duration;
```